test/model/recent_dm_conversations_test.dart

```diff
@@ -135,7 +143,42 @@ void main() {
         check(setupView()
           ..handleMessageEvent(MessageEvent(id: 1, message: message))
         ) ..map.deepEquals(expected.map)
-          ..sorted.deepEquals(expected.sorted);
+          ..sorted.deepEquals(expected.sorted)
+          ..latestMessagesByRecipient.deepEquals(expected.latestMessagesByRecipient);
+      });
```

Comment:
Added it in a separate commit, but `check(listenersNotified).isTrue();` instead of `check(listenersNotified).isFalse();`.

Reply:
Ah, hmm. In this case it's unnecessary to notify listeners, and slightly undesirable, since nothing in the data structure actually changes, right? Quoting the comment in `handleMessageEvent` in the case that this test exercises:

```dart
      // The conversation already has a newer message.
      // This should be impossible as long as we only listen for messages coming
      // through the event system, which sends events in order.
      // Anyway, do nothing.
```

I think it's not really _bad_ to notify listeners. This case isn't even supposed to be exercised at all, currently, and it may rarely get exercised after #650 in certain race conditions. But I don't think we should have a check that expects the slightly undesirable behavior.

Since the test caught my attention as possibly missing something, though, it might be helpful to add something to it, so the next reader doesn't have to wonder the same thing. Maybe a commented-out check; something like this:

```dart
        // (listeners are notified unnecessarily, but that's OK)
        // check(listenersNotified).isFalse();
```

Reply:
Separately, let's add a `..latestMessagesByRecipient.deepEquals` check to this test, in the commit that introduces `latestMessagesByRecipient`.